skytap/resource_skytap_vm_test.go

```diff
@@ -527,6 +421,39 @@ func TestAccCasandra(t *testing.T) {
 	})
 }
 
+func testAccCheckSkytapExternalPorts(t *testing.T, vmName string, count string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rsVM, err := getResource(s, vmName)
+		if err != nil {
+			return err
+		}
+		assert.Equal(t, count, rsVM.Primary.Attributes["service_ports.%"], "empty map entry")
+		assert.Equal(t, count, rsVM.Primary.Attributes["service_ips.%"], "empty map entry")
+
+		return nil
+	}
+}
+
+func setupEnvironmentWithKeys(templateKey string, templateIDFallback string, vmKey string, vmIDFallback string) (string, string, string) {
+	templateID := getEnv(templateKey, templateIDFallback)
+	vmID := getEnv(vmKey, vmIDFallback)
+	newEnvTemplateID := getEnv("SKYTAP_TEMPLATE_NEW_ENV_ID", templateID)
+	return templateID, vmID, newEnvTemplateID
+}
+
+func setupEnvironment(templateIDFallback string, vmIDFallback string) (string, string, string) {
+	return setupEnvironmentWithKeys("SKYTAP_TEMPLATE_ID", templateIDFallback, "SKYTAP_VM_ID", vmIDFallback)
+}
+
+func getEnv(key, fallback string) string {
```

Comment:
why in this file should be part of util?

Reply:
This is only used for testing.